smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -242,8 +255,18 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
         );
         lastNonceSubmitted = claimData.nonce;
 
-        // if we are double pegging, then we are going to need to get the address on this chain
-        address tokenAddress = claimData.doublePeg ? sourceAddressToDestinationAddress[claimData.tokenAddress] : claimData.tokenAddress;
+        // if we are double pegging AND we don't control the token, we deploy a new smart contract
+        address tokenAddress;
+        if(claimData.doublePeg && !_isManagedToken(claimData.tokenAddress)) {
+            string memory name;
+            string memory symbol;
+            uint8 decimals;
+            (name, symbol, decimals) = BridgeBank(bridgeBank).getTokenData(claimData.tokenAddress);
```

Comment:
If the token doesn't exist on this chain, how are we going to be able to get this data from the bridgebank? This doesn't make sense.

Reply:
You're absolutely right. Things are starting to make more sense in my mind. I'll rethink this flow.